response_builder.go

```diff
@@ -84,15 +70,47 @@ func withDefaultErrorHandler(next http.Handler) http.Handler {
 
 func withCombinedFragments(s *Server) http.Handler {
 	return http.HandlerFunc(func(rw http.ResponseWriter, r *http.Request) {
+		route := RouteFromContext(r.Context())
 		results := multiplexer.ResultsFromContext(r.Context())
 
 		if results != nil && results.Error() == nil {
 			resBuilder := newResponseBuilder(*s, rw)
-			resBuilder.SetLayout(results.Results()[0])
-			resBuilder.SetFragments(results.Results()[1:])
+			resBuilder.SetFragments(route, results.Results())
 			elapsed := time.Since(startTimeFromContext(r.Context()))
 			resBuilder.SetDuration(elapsed.Milliseconds())
 			resBuilder.Write()
 		}
 	})
 }
+
+func stitch(structure stitchStructure, results map[string]*multiplexer.Result) []byte {
+	childContent := make(map[string][]byte)
+
+	for _, childBuild := range structure.DependentStructures() {
+		childContent[childBuild.ReplacementID()] = stitch(childBuild, results)
+	}
+
+	self := results[structure.Key()].Body
+
+	// handle edge fragments
+	if len(childContent) == 0 {
+		return self
+	}
+
+	for replacementKey, content := range childContent {
+		directive := []byte(fmt.Sprintf("<viewproxy-fragment id=\"%s\"/>", replacementKey))
```

Comment:
I went with this as the replacement directive format since it's HTML like, but also sets us up for success if we decide to parse HTML in the future.

Reply:
Yeah, I like this id based approach, should we precompute this directive and reuse it across invocations you think?

So instead of map of replacement key to content it would be directive to content?

Reply:
I like the idea, but if I'm understanding it correctly I'm not sure we can with the current implementation since the replacement directive is relative to the parent, not absolute like the `.Key()`. (e.g. `header`, not `root.header`)

If we have fragment definitions like:

```go
viewProxyServer.Get("/hello/:name", fragment.Define(
	"/layouts/test_layout", fragment.WithoutValidation(), fragment.WithChildren(fragment.Children{
		"header": fragment.Define("/header/:name"),
		"body":   fragment.Define("/body/:name", fragment.WithChild("header", fragment.Define("/body/:name/header")),
	}),
))
```

Since both root and `body` have a `header`, both will have the same replacement directive of `<viewproxy-fragment id="header"/>`.